src/jobs/helm-lint.yaml

```diff
@@ -0,0 +1,24 @@
+---
```

Comment:
Do we need to make that a separate job? Can we include this in existing one? Where do you want to use it?

Reply:
There isn't really any existing job where this can go... The current Helm related jobs will package and publish Helm chart. The idea of this job is to do linting (and in future testing) of the Helm chart before we publish. So for example when PR is open, this will lint the Helm chart before it gets merged.

Reply:
How about making this part of push-to-app-catalog job then? We should have it everywhere.

Reply:
The way it's currently configured means that if I add it to push-to-app-catalog, it will only run if a tag is created (or w want to publish the Helm package). This isn't really what I think we want, we want to test before we commit to trying to create a package.

Reply:
push-to-app-catalog runs on every commit and tag. push-to-app-collection runs only on tag.

I'd say we want to test before pushing the chart to the catalog. Before that the chart is only built locally in the CI. Having the command included in existing job ensures that it's run once the architect orb version is bumped in the configuration.

Reply:
Maybe I've misunderstood the purpose of the jobs, but looking at the Kong config (done before I was around) and Nginx app, both only run `push-to-app-catalog` for `^v.*` tags:

https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/.circleci/config.yml#L92
https://github.com/giantswarm/kong-app/blob/master/.circleci/config.yml#L14

I agree that we want to use these tests are run before pushing, and putting them in the existing job is good. But I also think that we should run these tests for PR and not push. Would it right to create the jobs (in this PR) and also include it in existing jobs?

Reply:
This is CircleCI weirdness. Jobs are by default run for all branches and no tags. So to enable building _also_ for tags you have to add this filter. To build only on tags you'd have to do this:

https://github.com/giantswarm/aws-operator/blob/1e62a0b7dde996640f9f01832526ce284ee847ad/.circleci/config.yml#L118-L124

Reply:
Ah ok, that makes sense now :)

I am cool to add this to existing job, I'll update branch and PR. Thanks

Reply:
@kopiczko Implemented your suggestion, I have removed `helm-lint` job and incorporated the steps into `pust-to-app-catalog`. The change I had to make was to move template step, as it needs to run early for linting to work. Thanks